src/fw/services/common/bluetooth/bluetooth_ctl.c

```diff
@@ -96,12 +96,13 @@ static void prv_comm_start(void) {
   PBL_LOG(LOG_LEVEL_INFO, "BLE HRM sharing prefs: is_enabled=%u",
           config->is_hrm_supported_and_enabled);
 #endif
+  bt_persistent_storage_register_existing_ble_bondings();
+
```

Comment:
I think this will go wrong on the legacy Dialog driver that eventually got added to Google upstream for silk.  Maybe rename the `bt_driver_handle_host_added_bonding` API to `bt_driver_handle_host_added_bonding_early` (or make something in the NimBLE driver, like, `#define BLUETOOTH_CAN_HANDLE_EARLY_BLE_BONDING` and check for that in an `ifdef` here?) just to make sure that this errors obviously, rather than subtly, if someone tries to bring that back to life?

Reply:
added a define, so that we'll load store earlier or later depending on the stack requirements. I doubt that the Dialog stuff gets ported, though, it's a particularly weird/hacky architecture. If we could get a Dialog firmware with HCI, then using NimBLE host, maybe.

Reply:
I also think it's unlikely the Silk Dialog code will be resurrected. I added it to the OSS release since as weird as it is, a reference impl seemed better than no reference impl.

In practice I think once C2D is a thing, few folks will care about Pebble 2. Snowy/Spalding are a bit different because they're not being replaced directly, so they're more valuable to keep alive.